### PR TITLE
RESTX Rename

### PR DESCRIPTION
**Pull request by SteadBytes**
Addresses https://github.com/python-restx/flask-restx/issues/1

TODO:

- [x] Rename references to Flask-RESTPlus, flask-restplus e.t.c to Flask-RESTX, flask-restx e.t.c
- [x] Remove Flask-RESTPlus logo
- [x] Add new Flask-RESTX logo
- [x] Remove `AUTHORS.rst`
- [x] Remove `CHANGELOG.rst`
- [x] Update LICENCE
  - [x] Remove Akamai
  - [x] Add another `Modifed work Copyright` line (see questions)
- [x] Update `setup.py` `author` and `author_email` (see questions)

Questions:
- Which name should go into `setup.py` as the author and into the licence copyright? Are the other maintainers happy for my name  to be used here (as I'm making these changes)?

### CODE REVIEW

**Comment by apryor6** [comment]
Woo hoo for moving forward!

**Comment by andreykurilin** [comment]
> Which name should go into setup.py as the author and into the licence copyright? Are the other maintainers happy for my name to be used here (as I'm making these changes)?

python-restx team?

**Comment by ziirish** [comment]
Agreed, "python-restx" is good as author as from now on, this is a community driven project.

**Comment by SteadBytes** [comment]
That's fine by me, I just want to check if "python-restx team` is valid as the copyright line in the licence i.e. `Modified work Copyright (c) 2014 python-restx team`. Is anyone (more knowledgeable of licencing minutia than myself) able to comment on this?

**Comment by SteadBytes** [comment]
We also need an `author_email` for `setup.py` - any suggestions?

**Comment by andreykurilin** [comment]
@SteadBytes actually, only 3 fields are required: `name`,  `version`, and `packages`. We can just drop `author_email`.

Also, based on first clause of BSD license I'm not sure that we can remove `Akamai` copyright: `Redistributions of source code must retain the above copyright notice`. I guess the only available "workaround" is to revert an original [commit](https://github.com/python-restx/flask-restx/commit/b8ef37083747a76b3bdd387fe57edd47ca9214db) that adds copyright

**Comment by SteadBytes** [comment]
@andreykurilin `author_email` is required if `author` is specified, so we either have both or neither.

I defer to @j5awry on the Akamai licence bit, but he stated that he never made any copyrightable commits to the code whilst working at Akamai and won't be moving forward either.

**Comment by andreykurilin** [comment]
@SteadBytes 
> `author_email` is required if `author` is specified, so we either have both or neither.

oops, I did not know this. But after googling, it looks like not true anymore - https://github.com/python/cpython/pull/17388 (setuptools->distutils->cpython). I guess we can specify invalid email for now (if there are no other options) and remove it as soon as new python packages will be available.

> I defer to @j5awry on the Akamai licence bit, but he stated that he never made any copyrightable commits to the code whilst working at Akamai and won't be moving forward either.

yeah, but technically current change looks like 'forking with loss of one of the original authors'. At least, it is better to do it separately.

**Comment by SteadBytes** [comment]
@andreykurilin 
Ah yeah, that's change is quite recent - on my Python 3.7 install `setup.py check`  throws a warning:

```
python3 setup.py check
/usr/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'dev_require'
  warnings.warn(msg)
/usr/lib/python3/dist-packages/setuptools/dist.py:474: UserWarning: Normalizing '0.13.1.dev' to '0.13.1.dev0'
  normalized_version,
running check
warning: check: missing meta-data: if 'author' supplied, 'author_email' must be supplied too
```

As long as it doesn't affect building the project (i.e. it's only affect is showing that warning) then I'm happy to remove it - good spot thankyou!

I see your point, I will wait for @j5awry to confirm how to proceed with this it was originally introduced by/for him. The commit message has a clear explanation of why the change is taking place so as long as the explanation is actually valid then I think we're ok. That said, I'll make sure to have confirmation from @j5awry before proceeding further :+1:

File: README.rst [hunk 1]
```diff
@@ -2,63 +2,63 @@
...
     :alt: License
-.. image:: https://img.shields.io/pypi/pyversions/flask-restplus.svg
-    :target: https://pypi.org/project/flask-restplus
+.. image:: https://img.shields.io/pypi/pyversions/flask-restx.svg
+    :target: https://pypi.org/project/flask-restx
     :alt: Supported Python versions
 .. image:: https://badges.gitter.im/Join%20Chat.svg
-   :alt: Join the chat at https://gitter.im/noirbizarre/flask-restplus
-   :target: https://gitter.im/noirbizarre/flask-restplus?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
+   :alt: Join the chat at https://gitter.im/python-restx/flask-restx
```
Comment (andreykurilin) [inline README.rst]: wrong link https://gitter.im/python-restx/community

Comment (SteadBytes) [reply README.rst]: Good spot thank you, should be fixed now :+1:

**Comment by andreykurilin** [comment]
LGTM :) Thanks for pushing this forward!

**Comment by j5awry** [comment]
Legally, it's fine. There were 0  commits done with Akamai attached to it. Akamai _normally_ only cared about commit level (if i made a commit, add in "akamai copyright"). They normally don't do a top level.

In fact, doing a giant top level with a bunch of entries is highly irregular.

**Comment by j5awry** [comment]
Essentially, Akamai can't claim copyright when no copyrightable work by an Akamai employee is done. If you'd all rather, i can push the specific commit and have it in the commit message.

**Comment by andreykurilin** [comment]
I do not pretend to be a legal expert and I guess Akamai is not offended here, but it would be nice to save all this context into the history of the project. That is why I raised this question.

Please accept my apologies, @SteadBytes, I used to review PRs on GitHub where people forgot to add proper commit messages to all commits. Thanks to add a good commit message for a change to LICENCE!

**Comment by ziirish** [comment]
The tests are failing due to the lack of `author_email`:

```
warning: Check: missing meta-data: if 'author' supplied, 'author_email' must be supplied too
```

I tried locally with a dummy `author_email` and the tests succeed.

**Comment by andreykurilin** [comment]
@ziirish I guess we can modify https://github.com/python-restx/flask-restx/blob/master/tasks.py#L160 with `warn=False` to avoid CI failures. Another thing, as you mentioned, is setting author_email, but there is no team specific email yet...

**Comment by SteadBytes** [comment]
Thanks for clarification @j5awry - I'll leave the licence changes in :+1: No need to apologise @andreykurilin, one cannot be too careful when it comes to licencing!

Since the warning for `author_email` has been officially removed as @andreykurilin helpfully pointed out, I'm happy to adjust https://github.com/python-restx/flask-restx/blob/master/tasks.py#L160 to either: remove it entirely or make it output the warning but *not fail* the build. It's also somewhat confusingly named `readme_check` when it's testing `setup.py` :thinking: 

Of course the other option is to include the email, I'm *not* in favour of using a placeholder email. I see several options:
1. Use one of our own email addresses
2. Create a team email account (i.e. a Gmail account `python-restx@gmail.com`) and share access amongst the core team
3. Similar to point 2 but done 'properly' using a domain (i.e. G Suite)
    - **Major downside** for this is cost, hence I'm not in favour

**Comment by ziirish** [comment]
I'm on mobility atm, but I think I tested the underlying command `python setup.py test` and it doesn't return 0So I'm not sure patching the task as pointed earlier will resolve anything.There is another warning though. It suggests that `python setup.py test` has been deprecated in favor of `twine test dist/*`I tried that command and it doesn't complain about the lack of email. But this requires us to build at least the sdist first.Now I still think the name/description is valid because the purpose of this test is to check whether the documentation is renderable on Pypi.

**Comment by SteadBytes** [comment]
Ok awesome thanks @ziirish , looks like no `author_email` will be fine :+1:

**Comment by SteadBytes** [comment]
Brilliant, thanks for adding the logo @ziirish :smile: I'll give it one more check over, fixup the commits and then merge :+1:

**Pull request merged**